MetaMask/metamask-extension
# feat(21068): remove action queue for new MV3

## **Description**
For our purposes at MetaMask, one of the most significant differences between manifest v3 and v2 extension implementations was that the background process runs on a service worker that restarts every 5 minutes. However, [we recently learned](https://consensys.slack.com/archives/C02DUDT2WTU/p1693402457225629) that Chrome dropped that requirement, which means the mv3-specific code still needs to persist in the service worker state, but the implementation can be simplified from what we have previously implemented.
To remove the action queue functionality, we can essentially un-deprecate callBackgroundMethod and simplify it as well as submitRequestToBackground.

We can also simply call background.<bckgMethod>, but since these methods use callbacks, it's still useful to keep a wrapper function around the background methods that promisifies them.

Reference MR which added action queue previously:
https://github.com/MetaMask/metamask-extension/pull/15337

## **Manual testing steps**

1. Build the extension through the manifest v3 file using yarn dist:mv3 or yarn start:mv3
2. Load the extension and perform a few user actions such as adding a new account.
3. Wait 5 minutes (the previous limit for service worker lifecycle.
4. Perform a few user actions such as adding a new account
5. The actions should be performed and the app should not crash

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

Implements: https://github.com/MetaMask/metamask-extension/issues/21068

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
